generated/nidaqmx/base_interpreter.py

```diff
@@ -0,0 +1,1630 @@
+# Do not edit this file; it was automatically generated.
+from nidaqmx.constants import (
+    ACExcitWireMode, AccelChargeSensitivityUnits, AccelSensitivityUnits,
+    AccelUnits, AcquisitionType, AngleUnits, AngularVelocityUnits,
+    BridgeConfiguration, BridgeElectricalUnits, BridgePhysicalUnits,
+    BridgeUnits, CJCSource, ChargeUnits, CountDirection,
+    CounterFrequencyMethod, CurrentShuntResistorLocation, CurrentUnits,
+    DigitalPatternCondition, EddyCurrentProxProbeSensitivityUnits, Edge,
+    EncoderType, EncoderZIndexPhase, ExcitationSource,
+    ForceIEPESensorSensitivityUnits, ForceUnits, FrequencyUnits, FuncGenType,
+    GpsSignalType, LVDTSensitivityUnits, LengthUnits, Level, LineGrouping,
+    Polarity, PressureUnits, RTDType, RVDTSensitivityUnits,
+    ResistanceConfiguration, ResistanceUnits, Signal, SignalModifiers, Slope,
+    SoundPressureUnits, StrainGageBridgeType, StrainGageRosetteType,
+    StrainUnits, TEDSUnits, TemperatureUnits, TerminalConfiguration,
+    ThermocoupleType, TimeUnits, Timescale, TorqueUnits, UnitsPreScaled,
+    VelocityIEPESensorSensitivityUnits, VelocityUnits, VoltageUnits,
+    WindowTriggerCondition1, WriteBasicTEDSOptions)
+
+class BaseInterpreter:
```

Comment:
Would there be value in making this an abstract base class? https://peps.python.org/pep-3119/

nimi-python doesn't have a base interpreter class and uses duck typing.

I think that a base interpreter class **with type annotations** would add value compared to duck typing.

@mshafer-NI Do you have any suggestions?

Reply:
If @mshafer-NI doesn't have a different suggestion, let's do this:
- Inherit from `collections.abc.ABC`
- Change the base class method bodies from `pass` to `raise NotImplementedError` (https://docs.python.org/3/library/exceptions.html#NotImplementedError)

Reply:
I chatted with @mshafer-NI offline and he said a base class with ABC is fine as long as the shared interface (public methods) constitutes > 90% of each class.

Reply:
I almost forgot:
- Decorate each base class method with `@abc.abstractmethod`

Reply:
Sure. I have inherited `abc.ABC` into `BaseInterpreter` and added decorator `@abc.abstractmethod` for all the functions.